File: tests/src/com.squareup.retrofit2/retrofit/2.5.0/src/test/java/com_squareup_retrofit2/retrofit/RetrofitTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_squareup_retrofit2.retrofit;

import okhttp3.ResponseBody;
import org.junit.jupiter.api.Test;
import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.http.GET;

import static org.assertj.core.api.Assertions.assertThat;

public class RetrofitTest {
    @Test
    void createsProxyForAnnotatedServiceInterface() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("https://example.com/")
                .build();

        SimpleService service = retrofit.create(SimpleService.class);

        assertThat(service).isInstanceOf(SimpleService.class);
    }

    @Test
    void eagerlyValidatesDeclaredServiceMethods() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("https://example.com/")
                .validateEagerly(true)
                .build();

        EagerService service = retrofit.create(EagerService.class);

        assertThat(service).isInstanceOf(EagerService.class);
    }

    public interface SimpleService {
        @GET("messages")
        Call<ResponseBody> messages();
    }

    public interface EagerService {
        @GET("users")
        Call<ResponseBody> users();
    }
}

File: tests/src/com.squareup.retrofit2/retrofit/2.5.0/src/test/kotlin/com_squareup_retrofit2/retrofit/RetrofitTest.kt
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_squareup_retrofit2.retrofit

import okhttp3.ResponseBody
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.http.GET

public class RetrofitTest {
    public interface ExampleService {
        @GET("users")
        public fun users(): Call<ResponseBody>
    }

    @Test
    public fun createsServiceProxyAndEagerlyValidatesDeclaredMethods(): Unit {
        val retrofit: Retrofit = Retrofit.Builder()
            .baseUrl("https://example.com/")
            .validateEagerly(true)
            .build()

        val service: ExampleService = retrofit.create(ExampleService::class.java)
        val call: Call<ResponseBody> = service.users()

        assertThat(service).isInstanceOf(ExampleService::class.java)
        assertThat(call.request().url().toString()).isEqualTo("https://example.com/users")
    }
}
